server/src/main/java/com/linecorp/centraldogma/server/internal/mirror/MirroringTask.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.server.internal.mirror;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+import java.io.File;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.centraldogma.server.command.CommandExecutor;
+import com.linecorp.centraldogma.server.mirror.Mirror;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+
+final class MirroringTask {
+
+    private static Iterable<Tag> generateTags(Mirror mirror) {
+        return ImmutableList.of(
+                Tag.of("direction", mirror.direction().name()),
+                Tag.of("remoteRepo", mirror.remoteRepoUri().toString()),
+                Tag.of("remoteBranch", firstNonNull(mirror.remoteBranch(), "")),
+                Tag.of("remotePath", mirror.remotePath()),
+                Tag.of("localRepo", mirror.localRepo().name()),
+                Tag.of("localPath", mirror.localPath()));
+    }
+
+    private final MeterRegistry meterRegistry;
+    private final Mirror mirror;
+    private final Iterable<Tag> tags;
+
+    MirroringTask(Mirror mirror, MeterRegistry meterRegistry) {
+        this.mirror = mirror;
+        this.meterRegistry = meterRegistry;
+        tags = generateTags(mirror);
+    }
+
+    private Counter counter(boolean success) {
+        return Counter.builder("mirror.result")
```

Comment:
How about creating success counter and failure counter only once when `GitMirror` is created and reusing it?

Reply:
I do acknowledge that registering a counter to a `meterRegistry` involves a `synchronized` call.
Having said this, I'm wondering if you had a location to define the counters in mind since both solution I can think of don't seem trivial 😅 

1) Bringing a reference to `meterRegistry` into `AbstractMirror`
2) Maintaining a separate mapping between `Mirror` -> `Counter`

Reply:
I realized that the mirror configuration could be changed at the next mirroring so we cannot create the `counter` once and reuse it.
Also, there's a concurrent hash map inside with meter ID key so the counter is not created every time this is called unless the configuration is changed.
That being said, let's keep it as is. Sorry about it. 😅